src/smif/model/sos_model.py

```diff
@@ -202,51 +215,37 @@ def make_dependency_graph(models):
         return dependency_graph
 
     @staticmethod
-    def get_model_sets_in_run_order(graph, max_iterations, convergence_relative_tolerance,
+    def get_model_sets_in_job_graph(graph, step_id, data_handle, max_iterations,
+                                    convergence_relative_tolerance,
                                     convergence_absolute_tolerance):
-        """Returns a list of :class:`Model` in a runnable order.
+        """Returns a JobGraph of :class:`Model` in a runnable shape.
 
         If a set contains more than one model, there is an interdependency and
         and we attempt to run the models to convergence.
 
         Returns
         -------
-        list
-            A list of `smif.model.Model` objects
+        networkx.DiGraph
+            A graph with `smif.model.Model` Model, `smif.data_layer.DataHandle`
+            Data Handle and `string` operation attributes.
         """
-        if networkx.is_directed_acyclic_graph(graph):
-            # topological sort gives a single list from directed graph, currently
-            # ignoring opportunities to run independent models in parallel
-            run_order = networkx.topological_sort(graph)
-
-            # list of Models (typically ScenarioModel and SectorModel)
-            ordered_sets = list(run_order)
+        job_graph = networkx.DiGraph()
```

Comment:
Instead of copying nodes from a dependency graph over to a job graph, could we just create the job graph directly from the model+dependency structure?